js/src/components/adaptive-form/adaptive-form-context.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * External dependencies
+ */
+import { createContext, useContext } from '@wordpress/element';
+
+/**
+ * @typedef {Object} InputProps
+ * @property {*} value Form value.
+ * @property {boolean} checked Form value converted to boolean.
+ * @property {*} selected Form value.
+ * @property {(value: Event<HTMLInputElement> | *) => void} onChange Function to handle onChange event.
+ * @property {() => void} onBlur Function to handle onBlur event.
+ * @property {'has-error' | undefined} className 'has-error' if the form value is invalid and marked as touched. `undefined` otherwise.
+ * @property {string | undefined | null} help The corresponding value in form `errors` if the form value is marked as touched. `null` otherwise.
+ */
+
+/**
+ * @typedef {Object} AdaptiveFormContextAdapter
+ * @property {boolean} isSubmitting `true` if the form is currently being submitted.
+ * @property {boolean} isSubmitted Set to `true` after the form is submitted. Initial value and during submission are set to `false`.
+ * @property { HTMLElement | null} submitter Set to the element triggering the `handleSubmit` callback until the processing of `onSubmit` is completed. `null` otherwise.
+ */
+
+/**
+ * @typedef {Object} AdaptiveFormContext
+ * @property {Object} values Form values.
+ * @property {Object} errors Object with key-value pairs representing errors for form values. Empty object if no errors.
+ * @property {Object} touched Object with key-value pairs representing form values is touched.
+ * @property {boolean} isValidForm `true` if form values pass the validation.
+ * @property {boolean} isDirty `true` after any of the form values is modified.
+ * @property {(name: string, value: *) => void} setValue Function to set a form value.
+ * @property {(name: string) => InputProps} getInputProps Function to get the corresponding input props by the name of a form value.
+ * @property {() => Promise<Object>} handleSubmit Function to trigger form submission.
+ * @property {(initialValues: Object) => void} resetForm Function to reset form with given initial values.
+ * @property {(currentTouched: Object) => void} setTouched Function to update the `touched` state.
```

Comment:
Is not clear for me how it work this. Do we send the props as true  in `currentTouched`? Or what exactly should be inside `currentTouched`?

Reply:
Thanks for catching this. Just found the `currentTouched` naming was not correct and another type was missed. Fixed in
https://github.com/woocommerce/google-listings-and-ads/pull/1794/commits/e0b8dc68d52acd78ca26356215a79285f38bf9d4 with some other adjustments to make it clearer.

Reply:
In my IDE at least it marks the new type a s invalid:

I guess we can use

` * @property {React.Dispatch<React.SetStateAction<Object>>}`

Reply:
Thanks for the better suggestion. Updated in https://github.com/woocommerce/google-listings-and-ads/pull/1794/commits/b055ba1b05f6634783e4fd8b57d54a6e0d806b01.